client/src/components/QandA/QuestionList.jsx

```diff
@@ -2,31 +2,77 @@
+import { answerFormatter } from './Utilities.js';
 
-const QuestionList = (props) => {
-  return (
-    <div className="container">
-      <p className="text-start fs-5 fw-light">
-      Questions and Answers
-      </p>
-      <div className="input-group mb-3">
-        <input type="text" className="form-control" placeholder="HAVE A QUESTION? SEARCH FOR ANSWERS..." aria-label="Recipient's username" aria-describedby="button-addon2"></input>
-        <button className="btn btn-outline-secondary" type="button" id="button-addon2">Filter Questions</button>
-        {/* Above button label will be replaced by a magnifying glass icon in later update FIX ME ^^ */}
-      </div>
-      <div className="row">
-        {props.data.results.map((question, index) => {
-          if (index < 2) {
-            return <QuestionListEntry key={index} data={question} currentProductName={props.currentProductName} answers={props.answerParser(question.answers)}/>;
-          }
-          if (index >= 2) {
-            // return <LoadMoreQuestions data={question} answers={props.answerParser(question.answers)} />; <-- add LoadMoreQuestions component
+class QuestionList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      questionsShown: 2,
+      searchBarInput: '',
+      questionList: this.props.data
+    };
+    this.loadMoreQuestions = this.loadMoreQuestions.bind(this);
+    this.handleSearchBarInputChange = this.handleSearchBarInputChange.bind(this);
+  }
+
+  loadMoreQuestions() {
+    let { questionsShown } = this.state;
+    questionsShown += 2;
+    this.setState({questionsShown});
+  }
+
+  handleSearchBarInputChange(e) {
+    this.setState({searchBarInput: e.target.value});
+  }
+
+  render() {
+    const questions = this.state.questionList.results;
+    return (
+      <div className="container" maxHeight="vh-100">
+        <p className="text-start fs-5 fw-light">
+        Questions and Answers
+        </p>
+
+        {/* SEARCH/FILTER BAR */}
+
+        <div className="input-group mb-3">
+          <input type="text" className="form-control" placeholder="HAVE A QUESTION? SEARCH FOR ANSWERS..." onChange={this.handleSearchBarInputChange}></input>
+          <span className="btn btn-outline-secondary"><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" className="bi bi-search" viewBox="0 0 16 16">
+            <path d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z" /></svg>
+          </span>
+        </div>
+
+        {/* QUESTION LIST FILTERED BY 'QUESTIONSSHOWN' AND SEARCH INPUT */}
+
+        <div className="row">
+          {questions
+            .filter((q, i) => { return i < this.state.questionsShown; })
+            .filter((q) => { return this.state.searchBarInput.length >= 3 ? q.question_body.includes(this.state.searchBarInput) : q; })
```

Comment:
I suggest changing 'q' to something more descriptive, like question.

Reply:
Thank you - yes, I got a little carried away with abbreviating...